torchgeo/datamodules/nccm.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Northeastern China Crop Map (NCCM) datamodule."""
+
+from typing import Any, Optional, Union
+
+import kornia.augmentation as K
+from matplotlib.figure import Figure
+
+from ..datasets import NCCM, BoundingBox, Sentinel2
+from ..samplers import GridGeoSampler, RandomBatchGeoSampler
+from ..transforms import AugmentationSequential
+from .geo import GeoDataModule
+
+
+class NCCMSentinel2DataModule(GeoDataModule):
+    """LightningDataModule implementation for the NCCM and Sentinel2 datasets.
+
+    Uses the train/val/test splits from the dataset.
+    """
+
+    def __init__(
+        self,
+        batch_size: int = 64,
+        patch_size: Union[int, tuple[int, int]] = 256,
+        length: Optional[int] = None,
+        num_workers: int = 0,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize a new NCCMSentinel2DataModule instance.
+
+        Args:
+            batch_size: Size of each mini-batch.
+            patch_size: Size of each patch, either ``size`` or ``(height, width)``.
+            length: Length of each training epoch.
+            num_workers: Number of workers for parallel data loading.
+            **kwargs: Additional keyword arguments passed to
+                :class:`~torchgeo.datasets.NCCM` (prefix keys with ``nccm_``) and
+                :class:`~torchgeo.datasets.Sentinel2`
+                (prefix keys with ``sentinel2_``).
+        """
+        self.nccm_kwargs = {}
+        self.sentinel2_kwargs = {}
+        for key, val in kwargs.items():
+            if key.startswith("nccm_"):
+                self.nccm_kwargs[key[5:]] = val
+            elif key.startswith("sentinel2_"):
+                self.sentinel2_kwargs[key[10:]] = val
```

Comment:
If this starts to become a common kwarg parsing approach, we can add a utility function to avoid code duplication. But right now it's only used in 2 places.